### PR TITLE
#807 Remove EqualsAvoidNull check from Qulice 

### PR DESCRIPTION
**Pull request by kanghj**
#807.

Removed check, and added test.

### CODE REVIEW

**Comment by mkordas** [comment]
@krzyk can I review this PR to speed things up for @kanghj?

**Comment by krzyk** [comment]
@davvd assign @mkordas

**Comment by mkordas** [comment]
@kanghj I'll check it soon :)

File: qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java [hunk 1]
```diff
@@ -705,6 +705,17 @@ public void allowsDiamondOperatorUsage() throws Exception {
     }
 
     /**
+     * CheckstyleValidator can allow usage of string literals on either sides.
+     * E.g. both txt.equals("contents") and "contents".equals(txt) is valid.
```
Comment (mkordas) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @kanghj minor: should be `are valid`

Comment (mkordas) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @kanghj also, can you wrap the code in `{@code ...}` tags?

**Comment by mkordas** [comment]
@kanghj two minor comments from me

**Comment by kanghj** [comment]
@mkordas, this is ready to be reviewed again

**Comment by davvd** [comment]
@kanghj Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
> @davvd assign @mkordas 

@krzyk too many tasks already active in the project (2), can't have more

**Comment by mkordas** [comment]
@kanghj thanks, fixes look good to me

**Comment by mkordas** [comment]
@krzyk do you know which 2 tasks are assigned? I don't see anyhing... Maybe we need to try again?

**Comment by krzyk** [comment]
@mkordas there is #798 and #801

**Comment by kanghj** [comment]
@davvd, any chance of this proceeding soon?

**Comment by davvd** [comment]
> @davvd, any chance of this proceeding soon?

@kanghj will try our best

**Comment by kanghj** [comment]
@davvd, @yegor256, I am not paid, so this shouldn't affect the budget too much.

**Comment by krzyk** [comment]
@kanghj but there has to be a budget for reviewer

**Comment by krzyk** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job is not in WBS, won't close the order

**Pull request merged**